internal/services/paymentdispatchers/circle_payment_dispatcher.go

```diff
@@ -56,19 +63,25 @@ var _ PaymentDispatcherInterface = (*CirclePaymentDispatcher)(nil)
 
 func (c *CirclePaymentDispatcher) sendPaymentsToCircle(ctx context.Context, sdpDBTx db.DBTransaction, circleWalletID string, paymentsToSubmit []*data.Payment) error {
 	for _, payment := range paymentsToSubmit {
+		// 0. Ensure the recipient is ready
```

Comment:
Totally agreed! I prefer to handle this in my next PR since this one has already become too big.

I've added a "Pending" section to this PRs description where I describe this issue as something that needs to be solved and I'm preparing a PR with the fix.

Reply:
Thank you!